### PR TITLE
Miscellaneous changes

### PR DESCRIPTION
**Pull request by bwohlberg**
* Add new example demonstrating ℓ1-TV denoising
* Add support for generic loss function
* Fix handling of `scale` parameter in `SVMBIRExtendedLoss.__init__`
* Clean up loss tests
* Resolve docs build warning
* Address some style violations in `radon_svmbir`

### CODE REVIEW

**Review by Michael-T-McCann — APPROVED** [review]
Looking good!

File: examples/scripts/denoise_l1tv_iso_admm.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+This example demonstrates impulse noise removal via ℓ1 total variation
+:cite:`alliney-1992-digital` :cite:`esser-2010-primal` (Sec. 2.4.4)
+(i.e. total variation regularization with an ℓ1 data fidelity term),
+minimizing the functional
+
+  $$\mathrm{argmin}_{\mathbf{x}} \;  \| \mathbf{y} - \mathbf{x}
+  \|_1 + \lambda \| C \mathbf{x} \|_1 \;,$$
+
+where $\mathbf{y}$ is the noisy image, $C$ is a 2D finite fifference
```
Comment (tbalke) [inline examples/scripts/denoise_l1tv_iso_admm.py:19]: "fifference"

File: examples/scripts/denoise_l1tv_iso_admm.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+"""
+Create a ground truth image and impose salt & pepper noise to create a
+noisy test image.
+"""
+N = 256  # image size
+phantom = SiemensStar(16)
+x_gt = snp.pad(discrete_phantom(phantom, 240), 8)
+x_gt = jax.device_put(x_gt)  # convert to jax type, push to GPU
+x_gt = 0.5 * x_gt / x_gt.max()
```
Comment (tbalke) [inline examples/scripts/denoise_l1tv_iso_admm.py:42]: Seems cleaner if you first scale and then put it on the device

File: examples/scripts/denoise_tv_iso_admm.py [hunk 3]
```diff
@@ -9,10 +9,11 @@
...
 This example compares denoising via isotropic and anisotropic total
-variation (TV) regularization. It solves the denoising problem
+variation (TV) regularization :cite:`rudin-1992-nonlinear`
+:cite:`goldstein-2009-split`. It solves the denoising problem
 
   $$\mathrm{argmin}_{\mathbf{x}} \; (1/2) \| \mathbf{y} - \mathbf{x}
-  \|^2 + \lambda R(\mathbf{x}) \;,$$
+  \|_2^2 + \lambda R(\mathbf{x}) \;,$$
 
 where $R$ is either the isotropic or anisotropic TV regularizer.
```
Comment (tbalke) [inline examples/scripts/denoise_tv_iso_admm.py:18]: perhaps use lowercase for the regularizer since it is not an operator?

File: examples/scripts/denoise_l1tv_iso_admm.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+ℓ1 Total Variation (ADMM)
+=========================
+
+This example demonstrates impulse noise removal via ℓ1 total variation
+:cite:`alliney-1992-digital` :cite:`esser-2010-primal` (Sec. 2.4.4)
+(i.e. total variation regularization with an ℓ1 data fidelity term),
+minimizing the functional
+
+  $$\mathrm{argmin}_{\mathbf{x}} \;  \| \mathbf{y} - \mathbf{x}
+  \|_1 + \lambda \| C \mathbf{x} \|_1 \;,$$
```
[the remark this answers is not in the document]
Comment (bwohlberg) [reply examples/scripts/denoise_l1tv_iso_admm.py:17]: That would make sense, but I couldn't find an instance of this in any of the other example scripts.

File: examples/scripts/denoise_l1tv_iso_admm.py (hunk 1 above)
Comment (bwohlberg) [reply examples/scripts/denoise_l1tv_iso_admm.py:19]: Good catch. Fixed.

File: examples/scripts/denoise_l1tv_iso_admm.py (hunk 2 above)
Comment (bwohlberg) [reply examples/scripts/denoise_l1tv_iso_admm.py:42]: Agreed. Done.

File: examples/scripts/denoise_tv_iso_admm.py (hunk 3 above)
Comment (bwohlberg) [reply examples/scripts/denoise_tv_iso_admm.py:18]: That would make sense, but this usage is scattered across multiple examples, so it may be better to fix it as part of a separate PR.

File: scico/examples.py [hunk 5]
```diff
@@ -291,3 +291,29 @@ def create_circular_phantom(
         img = img.at[dist_map < r].set(val)
 
     return img
+
+
+def spnoise(img: Array, frc: float, smn: float = 0.0, smx: float = 1.0) -> Array:
```
Comment (tbalke) [inline scico/examples.py:296]: what do smn and smx stand for? Perhaps longer, more descriptive names would be nicer.

File: scico/linop/radon_svmbir.py [hunk 6]
```diff
@@ -57,8 +57,8 @@ def __init__(
             angles: Array of projection angles in radians, should be
                 increasing.
             num_channels: Number of pixels in the sinogram.
-            center_offset: Position of the detector center relative to the center-of-rotation,
-                in units of pixels
+            center_offset: Position of the detector center relative to
+                the center-of-rotation, in units of pixels
```
Comment (tbalke) [inline scico/linop/radon_svmbir.py:61]: Hyphenation necessary in center-of-rotation?

Comment (bwohlberg) [reply scico/linop/radon_svmbir.py:61]: No. Fixed.

File: scico/loss.py [hunk 7]
```diff
@@ -51,27 +51,34 @@ def __init__(
         self,
         y: Union[JaxArray, BlockArray],
         A: Optional[Union[Callable, operator.Operator]] = None,
-        scale: float = 0.5,
+        l: Optional[functional.Functional] = None,
```
Comment (tbalke) [inline scico/loss.py:54]: I'm on the fence about naming it l. (Yes that is an l and not an I. Also it is an `l` and not a `1`). See the problem?

File: scico/examples.py (hunk 5 above)
Comment (bwohlberg) [reply scico/examples.py:296]: Agreed. Fixed.

File: scico/loss.py [hunk 8]
```diff
@@ -80,7 +87,37 @@ def __call__(self, x: Union[JaxArray, BlockArray]) -> float:
...
+
+    def prox(
+        self, v: Union[JaxArray, BlockArray], lam: float, **kwargs
+    ) -> Union[JaxArray, BlockArray]:
+        r"""Scaled proximal operator of loss function.
+
+        Evaluate scaled proximal operator of this loss function, with
+        scaling :math:`\lambda` = `lam` and evaluated at point
+        :math:`\mb{v}` = `v`. If :meth:`prox` is not defined in a derived
+        class, and if operator :math:`A` is the identity operator, the
```
Comment (tbalke) [inline scico/loss.py:104]: inserting a `then` between `operator` and `the` makes it easier to read in my opinion.

File: scico/loss.py (hunk 7 above)
Comment (bwohlberg) [reply scico/loss.py:54]: This occurred to me too, but all the other parameter names set the pattern of matching the labels used in the associated equation. Any suggestions for alternatives, which would probably also need to be used in the equation?

File: scico/loss.py (hunk 8 above)
Comment (bwohlberg) [reply scico/loss.py:104]: Agreed. Done.

File: scico/test/test_loss.py [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        self.y, key = randn((n,), key=key, dtype=dtype)
+        self.v, key = randn((n,), key=key, dtype=dtype)  # point for prox eval
+        scalar, key = randn((1,), key=key, dtype=dtype)
+        self.scalar = scalar.copy().ravel()[0]
+
+    def test_generic_squared_l2(self):
+        A = linop.Identity(input_shape=self.y.shape)
+        l = functional.SquaredL2Norm()
+        L0 = loss.Loss(self.y, A=A, l=l, scale=0.5)
+        L1 = loss.SquaredL2Loss(y=self.y, A=A)
```
Comment (tbalke) [inline scico/test/test_loss.py:39]: perhaps call then `f`, `f0`, `f1`? We use `L` for the Lipschitz constant or upper case for operators.

Comment (bwohlberg) [reply scico/test/test_loss.py:39]: Agreed, but `L` is already widely used in the same way in all the other existing tests in the same file. I'm inclined to leave addressing this for a separate PR.

File: scico/loss.py (hunk 7 above)
Comment (tbalke) [reply scico/loss.py:54]: Perhaps we want to make it `L` for now and then this issue can be grouped in a combined new PR `Ambiguity in symbol L as Lipschitz constant or loss`. (See other comment below)

**Review by tbalke — APPROVED** [review]

File: examples/scripts/denoise_l1tv_iso_admm.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (bwohlberg) [reply examples/scripts/denoise_l1tv_iso_admm.py:17]: I'm going to defer addressing this to a later cleanup.

File: scico/loss.py (hunk 7 above)
Comment (bwohlberg) [reply scico/loss.py:54]: Change it to `f`.
